js/state/syncUtil.js

```diff
@@ -64,6 +64,59 @@ const applySiteRecord = (record, siteDetail, tag) => {
   }
 }
 
+const applySiteSettingRecord = (record) => {
+  // TODO: In Sync lib syncRecordAsJS() convert Enums into strings
+  const adControlEnum = {
+    0: 'showBraveAds',
+    1: 'blockAds',
+    2: 'allowAdsAndTracking'
+  }
+  const cookieControlEnum = {
+    0: 'block3rdPartyCookie',
+    1: 'allowAllCookies'
+  }
+  const appActions = require('../actions/appActions')
+  const objectId = new Immutable.List(record.objectId)
+  const category = CATEGORY_MAP[record.objectData].categoryName
+  const hostPattern = record.siteSetting.hostPattern
+  if (!hostPattern) {
+    throw new Error('siteSetting.hostPattern is required.')
+  }
+
+  let applySetting = null
+  switch (record.action) {
+    case writeActions.CREATE:
+    case writeActions.UPDATE:
+      applySetting = (key, value) => {
+        appActions.changeSiteSetting(hostPattern, key, value, false, true)
+      }
+      break
+    case writeActions.DELETE:
+      applySetting = (key, _value) => {
+        appActions.removeSiteSetting(hostPattern, key, false, true)
+      }
+      break
+  }
+  // First set the objectId if needed so we can access the existing objectj
+  let existingObject = this.getObjectById(objectId, category)
+  if (!existingObject) {
+    applySetting('objectId', objectId)
+    existingObject = this.getObjectById(objectId, category)
+  }
+  const existingObjectData = existingObject[1]
+  for (let key in record.siteSetting) {
+    if (key === 'hostPattern') { continue }
+    let value = record.siteSetting[key]
+    if (key === 'adControl') {
+      value = adControlEnum[value]
+    } else if (key === 'cookieControl') {
+      value = cookieControlEnum[value]
+    }
+    if (existingObjectData.get(key) === value) { continue }
```

Comment:
i think this needs to call `applySetting` anyway (instead of `continue`) when the action is DELETE. `null` may be a valid siteSetting